pkglistgen/tool.py

```diff
@@ -785,7 +918,27 @@ def update_and_solve_target(
                 for line in sorted(output):
                     f.write(line + '\n')
 
-        self.commit_package(product_dir)
+        if git_url:
+            if os.path.isfile(f"{self.productcompose_dir}/default.productcompose"):
+                logging.debug(subprocess.check_output(
+                    ['git', 'add', self.productcompose_dir, f"{self.productcompose_dir}/default.productcompose"],
+                    cwd=cache_dir, encoding='utf-8'))
+            if product_dir and os.path.isdir(product_dir):
+                logging.debug(subprocess.check_output(
+                    ['git', 'add', product_dir],
+                    cwd=cache_dir, encoding='utf-8'))
+            if os.path.isdir(self.productcompose_dir):
+                logging.debug(subprocess.check_output(
+                    ['git', 'add', self.productcompose_dir],
+                    cwd=cache_dir, encoding='utf-8'))
+
+            logging.debug(subprocess.check_output(
+                'git commit -m "Update by pkglistgen of openSUSE-release-tool"', cwd=cache_dir, shell=True, encoding='utf-8'))
```

Comment:
Shouldn't create a commit on dry-run

Reply:
thinking again, imho it should. dry-run should do really everything, exception modification on the server to keep difference as little as possible to non-dry run. So any local modification, including commit should be done.

Reply:
:shrug: